### PR TITLE
eralchemy v1.0.1 (new formula)

### PR DESCRIPTION
**Pull request by Alexis-benoist**
Added ERAlchemy

### CODE REVIEW

File: Library/Formula/eralchemy.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        system "python", *Language::Python.setup_install_args(libexec/"vendor")
+      end
+    end
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir[libexec/"bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
```
Comment (bfontaine) [inline Library/Formula/eralchemy.rb]: Please add a `test do` block.

**Comment by apjanke** [comment]
Audit failure:

```
== Library/Formula/eralchemy.rb ==
C: 38:  4: Final newline missing.
1 file inspected, 1 offense detected
Error: 1 problem in 1 formula
==> audit problems
eralchemy:
 * File should end with a newline
```

**Comment by Alexis-benoist** [comment]
Do you guys need a response from me?

File: Library/Formula/eralchemy.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  test do
+    system "#{bin}/eralchemy", "-v"
```
Comment (bfontaine) [inline Library/Formula/eralchemy.rb]: Can’t we write a test that does something more substantial than e.g. `-v`/`-h`?

Comment (Alexis-benoist) [inline Library/Formula/eralchemy.rb]: The test I want to write is the following:
1) Check version
2) Create a test SQLite database
3) Draw the ER diagram of this DB
4) Check that the diagram exists.

For 2), I don't know if I can call this [
https://github.com/Alexis-benoist/eralchemy/blob/master/tests/create_db.py](script)
directly (which will be packaged as a ressource), or if I need to include
it to the eralchemy cli.

2015-12-31 17:23 GMT+01:00 Baptiste Fontaine <notifications@github.com>:

Comment (bfontaine) [inline Library/Formula/eralchemy.rb]: To make it simpler you could use the example mentioned in the [README](https://github.com/Alexis-benoist/eralchemy#use-from-the-command-line-and-a-markdown-file) with a Markdown file, e.g.:

```rb
resource "newsmeme.er" do
  url "https://raw.githubusercontent.com/Alexis-benoist/eralchemy/v1.0.1/example/newsmeme.er"
  sha256 "..."
end
```

Comment (Alexis-benoist) [inline Library/Formula/eralchemy.rb]: I will do that!

**Comment by Alexis-benoist** [comment]
I wrote the [test](https://github.com/Alexis-benoist/formula-eralchemy/blob/master/eralchemy.rb) and it works and passes audits, what do you think about it? Is it good for PR?

**Comment by bfontaine** [comment]
@Alexis-benoist Please use `assert File.exist?("test_eralchemy.pdf")` and if you want to be perfect add a newline between the two resources definitions, between the last resource and the `def install`, and one above the `test do`. If not it’s ok :+1:

**Comment by Alexis-benoist** [comment]
Ok, I did it :)

**Comment by bfontaine** [comment]
Thank you @Alexis-benoist for your contribution to Homebrew; we appreciate it! :tada:

File: Library/Formula/eralchemy.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir[libexec/"bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+    system "#{bin}/eralchemy", "-v"
+    resource("er_example").stage do
+      system "#{bin}/eralchemy", "-i", "/Library/Caches/Homebrew/eralchemy--er_example-1.0.1.er", "-o", "test_eralchemy.pdf"
```
Comment (bfontaine) [inline Library/Formula/eralchemy.rb]: I didn’t notice that when reviewing but I fixed it when I pulled: the block is executed in a temporary directory with the resource in it so you can just use `"newsmeme.er"` for its path.

**Comment by Alexis-benoist** [comment]
Thank you!

**Pull request closed without merging**